tomcat/src/main/java/nextstep/jwp/controller/LoginController.java

```diff
@@ -17,50 +17,46 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class LoginController implements Controller {
+public class LoginController extends AbstractController {
 
     private static final String ACCOUNT = "account";
     private static final String PASSWORD = "password";
     private static final Logger log = LoggerFactory.getLogger(Application.class);
 
-    private final Manager manager;
+    private static final Controller INSTANCE = new LoginController();
 
-    public LoginController(final Manager manager) {
-        this.manager = manager;
+    public static Controller getInstance() {
+        return INSTANCE;
     }
 
-    @Override
-    public HttpResponse service(final HttpRequest httpRequest) {
-        if (httpRequest.isGetMethod()) {
-            return doGet(httpRequest);
-        }
-
-        return doPost(httpRequest);
+    private LoginController() {
     }
 
-    private HttpResponse doGet(final HttpRequest httpRequest) {
+    @Override
+    protected HttpResponse doGet(final HttpRequest httpRequest) {
```

Comment:
객체를 꺼내서 직접 처리하지 말고 객체에게 물어보세요.

```suggestion
    @Override
    protected HttpResponse doGet(final HttpRequest httpRequest) {
        if (httpRequest.hasSessionId()) {
            return HttpResponse.of(HttpStatusCode.OK, "/login.html");
        }

        return generateSuccessResponse();
    }
```